File: Julia/DecryptAffine_German.jl
include("ExtendedEuclidean.jl")

function DecryptAffine_German(plaintext_ascii_30, a, b)
    a_t = ExtendedEuclidean(30, a)["t"]
    return mod.((a_t * (plaintext_ascii_30 - 65) - b * a_t), 30) + 97
end

File: JavaScript/DecryptAffine_German.js
import { mod } from "./mod.js";
import { ExtendedEuclidean } from "./ExtendedEuclidean.js";

export function DecryptAffine_German(num) {
    // Shift back using y = ax + b
    var a = 7;
    var b = 5;
    var t = ExtendedEuclidean(30, a).t
    return mod(t * (num - 65) - b * t, 30) + 97;
};
